docker-compose.test.yaml

```diff
@@ -1,10 +1,12 @@
 version: '3'
 
-# This override file should be used when running automated tests so
-# that test output is saved to the host
+# This override file should be used to run automated tests in an
+# isolated container stack
 
 services:
   ffc-demo-web:
     command: npm run test
-    volumes:
-      - ./test-output:/usr/src/app/test-output
+    container_name: ffc-demo-web-test-${BUILD_NUMBER:-local}

```

Comment:
there was a discussion about including the PR number or branch (i.e. the container tag) to ensure builds are isolated between different PRs/master builds, is it worth adding that in this work?

Reply:
Yep, that definitely makes sense. Needs a bit of thought as we're not currently using a Jenkins integration that provides the PR number automatically.

I think we can set environment variables within the pipeline but it might be worth switching to a GitHub-specific branch source that lets us build specifically for PRs, rather than all branches.

Reply:
This would remove the need to obtain the PR value ourselves, but as the unique identifying string makes use of the branch or the PR number we still need to set an env var ourselves.

Reply:
I've updated to version 0.0.3 of the shared Jenkins library, which adds your change to include the PR number in the project prefix when running tests. I've also removed all the `container_name` usage from the docker-compose yamls.

I also removed the custom-named network in favour of using Docker Compose's default networking and added nodemon config to stop reloading the app on changes to `./test-output`.